pkg/authorization/api/conversion.go

```diff
@@ -0,0 +1,192 @@
+	}
+	var err error
+	if out.Subjects, err = convertOriginSubjects(in.Subjects); err != nil {
+		return err
+	}
+	out.RoleRef = convertOriginRoleRef(&in.RoleRef)
+	out.ObjectMeta = in.ObjectMeta
+	return nil
+}
+
+func convertOriginPolicyRule(in []PolicyRule) []rbac.PolicyRule {
+	rules := make([]rbac.PolicyRule, 0, len(in))
+	for _, rule := range in {
+		r := rbac.PolicyRule{ // AttributeRestrictions is lost, but our authorizor ignores that field now
+			APIGroups:       rule.APIGroups,
+			Verbs:           rule.Verbs.List(),
+			Resources:       rule.Resources.List(),
+			ResourceNames:   rule.ResourceNames.List(),
+			NonResourceURLs: rule.NonResourceURLs.List(),
+		}
+		rules = append(rules, r)
+	}
+	return rules
+}
+
+func convertOriginSubjects(in []api.ObjectReference) ([]rbac.Subject, error) {
+	subjects := make([]rbac.Subject, 0, len(in))
+	for _, subject := range in {
+		s := rbac.Subject{
+			Name:       subject.Name,
+			APIVersion: rbac.GroupName, // TODO what to use here?
+		}
+
+		switch subject.Kind {
+		case ServiceAccountKind:
+			s.Kind = rbac.ServiceAccountKind
+			s.Namespace = subject.Namespace
+		case UserKind, SystemUserKind:
+			s.Kind = rbac.UserKind
+		case GroupKind, SystemGroupKind:
+			s.Kind = rbac.GroupKind
+		default:
+			return nil, fmt.Errorf("invalid kind for origin subject: %q", subject.Kind)
+		}
+
+		subjects = append(subjects, s)
+	}
+	return subjects, nil
+}
+
+func convertOriginRoleRef(in *api.ObjectReference) rbac.RoleRef {
+	return rbac.RoleRef{
+		APIGroup: in.APIVersion,
+		Kind:     in.Kind, // TODO leave empty?
+		Name:     in.Name,
+	}
+}
+
+func Convert_rbac_ClusterRole_To_api_ClusterRole(in *rbac.ClusterRole, out *ClusterRole, _ conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Rules = convertRBACPolicyRules(in.Rules)
+	return nil
+}
+
+func Convert_rbac_Role_To_api_Role(in *rbac.Role, out *Role, _ conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Rules = convertRBACPolicyRules(in.Rules)
+	return nil
+}
+
+func Convert_rbac_ClusterRoleBinding_To_api_ClusterRoleBinding(in *rbac.ClusterRoleBinding, out *ClusterRoleBinding, _ conversion.Scope) error {
+	var err error
+	if out.Subjects, err = convertRBACSubjects(in.Subjects); err != nil {
+		return err
+	}
+	out.RoleRef = convertRBACRoleRef(&in.RoleRef, "")
+	out.ObjectMeta = in.ObjectMeta
+	return nil
+}
+
+func Convert_rbac_RoleBinding_To_api_RoleBinding(in *rbac.RoleBinding, out *RoleBinding, _ conversion.Scope) error {
+	var err error
+	if out.Subjects, err = convertRBACSubjects(in.Subjects); err != nil {
+		return err
+	}
+	out.RoleRef = convertRBACRoleRef(&in.RoleRef, in.Namespace)
+	out.ObjectMeta = in.ObjectMeta
+	return nil
+}
+
+func convertRBACSubjects(in []rbac.Subject) ([]api.ObjectReference, error) {
+	subjects := make([]api.ObjectReference, 0, len(in))
+	for _, subject := range in {
+		s := api.ObjectReference{
+			APIVersion: rbac.GroupName, // TODO what do we want here?
```

Comment:
This is empty.  You really should read the existing code that builds origin subject references.